modulo5/projeto-todolist/src/index.ts

```diff
@@ -0,0 +1,109 @@
+import connection from './connection';
+import app from './app';
+import { Request, Response } from 'express';
+import { v4 as generateId } from 'uuid';
+
+const id = generateId()
+
+
+app.post("/user", async (req: Request, res: Response) => {
+    try {
+        const { name, nickname, email } = req.body
+        if (!name || !nickname || !email) {
+            throw new Error("Parâmetro não enviado");
+        }
+        await connection.raw(`
+        INSERT INTO ToDoListUser (id, name, nickname, email)
+        VALUES(
+            "${generateId()}",
+            "${name}",
+            "${nickname}",
+            "${email}"
+        )
+        `)
+        res.status(201).send(`Usuário ${nickname} adicionado com sucesso!`);
+    } catch (error: any) {
+        res.status(500).send(error.message);
+    }
+
+})
+
+app.get("/user/:id", async (req: Request, res: Response) => {
+    const id: string = req.params.id;
+    try {
+        const idFromUser = await connection.raw(`
+        SELECT * FROM ToDoListUser
+        WHERE id = '${id}'
+     `);
+        res.status(200).send(idFromUser[0][0]);
+    } catch (error: any) {
+        res.status(500).send(error.message);
+    }
+});
+
+app.put("user/edit/:id"), async (req: Request, res: Response) => {
+    try {
+        const id = req.params.id
+        if (!id) { throw new Error("Parâmetro id não enviado") }
```

Comment:
Bom cuidado de se ter, mas, neste caso, esse id sempre vai existir! Porque o express vai chamar esta rota quando a requisição for do tipo 'user/id/qualquerCoisa'.